### PR TITLE
Impact - Add Label for Edge

### PR DESCRIPTION
**Pull request by tsmr**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

Impact - Add Label for Edges on Cytoscape display : 

image

### CODE REVIEW

File: js/impact.js [hunk 1]
```diff
@@ -836,6 +841,15 @@ var GLPIImpact = {
                     'target-distance-from-node': '0px',
                 }
             },
+            {
+                selector: "edge[label]",
+                css: {
+                    "label": "data(label)",
+                    "text-rotation": "autorotate",
+                    "text-margin-x": "0px",
+                    "text-margin-y": "0px"
```
Comment (cconard96) [inline js/impact.js:850]: Labels may look better with text-margin-y: -10px

**Review by orthagh — APPROVED** [review]
Ok for adding the feature (I didn't review the changes)

File: js/impact.js [hunk 2]
```diff
@@ -1334,6 +1363,14 @@ var GLPIImpact = {
                 selector       : 'node[hasITILObjects=1]',
                 onClickFunction: this.menuOnShowOngoing
             },
+            {
+                id             : 'editEdge',
+                content        : '<i class="fas fa-edit me-2"></i>' + __("Edge properties..."),
+                tooltipText    : _.unescape(__("Set name for this edge")),
```
Comment (cedric-anne) [inline js/impact.js:1369]: Special chars should be escaped, even on translations.
```suggestion
                content        : '<i class="fas fa-edit me-2"></i>' + _.escape(__("Edge properties...")),
                tooltipText    : _.escape(__("Set name for this edge")),
```

File: install/migrations/update_10.0.15_to_10.0.16/impactrelations.php
Comment (cedric-anne) [inline install/migrations/update_10.0.15_to_10.0.16/impactrelations.php:1]: This file should be moved to the 10.0.x->11.0.0 migrations directory.

**Review by cedric-anne — CHANGES_REQUESTED** [review]

File: install/migrations/update_10.0.15_to_10.0.16/impactrelations.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ * ---------------------------------------------------------------------
+ */
+
+/**
+ * @var \Migration $migration
+ */
+
+$migration->addField('glpi_impactrelations', 'name', "varchar(255) DEFAULT ''", [
+    'after'  => 'id',
+]);
```
Comment (AdrienClairembault) [inline install/migrations/update_10.0.15_to_10.0.16/impactrelations.php:42]:
```suggestion
$migration->addField('glpi_impactrelations', 'name', "string", [
    'after'  => 'id',
    'value' => '',
]);
```

We try to use the "official" fields format defined in `migration->fieldFormat()` if possible.
The `string` format would be fine here I think.

File: src/Impact.php [hunk 4]
```diff
@@ -1401,7 +1402,8 @@ private static function addEdge(
         string $key,
         CommonDBTM $itemA,
         CommonDBTM $itemB,
-        int $direction
+        int $direction,
+        $label
```
Comment (AdrienClairembault) [inline src/Impact.php:1406]:
```suggestion
        string $label
```

File: js/impact.js [hunk 5]
```diff
@@ -83,7 +83,7 @@ var GLPIImpact = {
     ACTION_EDIT_DEPENDS_COLOR           : 12,
     ACTION_EDIT_IMPACT_COLOR            : 13,
     ACTION_EDIT_IMPACT_AND_DEPENDS_COLOR: 14,
-
+    ACTION_EDIT_EDGE                    : 15,
```
Comment (AdrienClairembault) [inline js/impact.js:86]: I think we need to take this new action into account in the `undo` and `redo` functions.

**Review by AdrienClairembault — CHANGES_REQUESTED** [review]

**Review by AdrienClairembault — APPROVED** [review]
Looks good to me.

File: templates/impact/edit_edge_modal.html.twig [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+ #}
+
+<div id="edit_edge_dialog"  class="modal" tabindex="-1" role="dialog">
+    <div class="modal-dialog modal-dialog-centered" role="document">
+        <div class="modal-content">
+
+            {# Header #}
+            <div class="modal-header">
+                <h5 class="modal-title"> {{ __('Edit edge') }} </h5>
+                <button type="button" class="btn btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
```
Comment (cedric-anne) [inline templates/impact/edit_edge_modal.html.twig:41]:
```suggestion
                <button type="button" class="btn btn-close" data-bs-dismiss="modal" aria-label="{{ _('Close') }}"></button>
```

File: src/Impact.php [hunk 7]
```diff
@@ -1522,6 +1525,17 @@ public static function printEditCompoundDialog(): void
         TemplateRenderer::getInstance()->display('impact/edit_compound_modal.html.twig');
     }
 
+    /**
+     * Load the "edit edge" dialog
+     *
+     * @since 9.5
+     */
+    public static function printEditEdgeDialog()
```
Comment (cedric-anne) [inline src/Impact.php:1533]:
```suggestion
     */
    private static function printEditEdgeDialog(): void
```

**Review by cedric-anne — APPROVED** [review]
Seems OK, but I did not take time to test it.

File: templates/impact/edit_edge_modal.html.twig [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+ #}
+
+<div id="edit_edge_dialog"  class="modal" tabindex="-1" role="dialog">
+    <div class="modal-dialog modal-dialog-centered" role="document">
+        <div class="modal-content">
+
+            {# Header #}
+            <div class="modal-header">
+                <h5 class="modal-title"> {{ __('Edit edge') }} </h5>
+                <button type="button" class="btn btn-close" data-bs-dismiss="modal" aria-label="{{ _('Close') }}"></button>
```
Comment (cedric-anne) [inline templates/impact/edit_edge_modal.html.twig:41]:
```suggestion
                <button type="button" class="btn btn-close" data-bs-dismiss="modal" aria-label="{{ __('Close') }}"></button>
```

**Pull request merged**